src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.String.cs

```diff
@@ -14,7 +18,13 @@ internal partial class CborReader
         public byte[] ReadByteString()
         {
             CborInitialByte header = PeekInitialByte(expectedType: CborMajorType.ByteString);
-            int length = checked((int)ReadUnsignedInteger(header, out int additionalBytes));
+
+            if (header.AdditionalInfo == CborAdditionalInfo.IndefiniteLength)
```

Comment:
Does there need to be a state check here that this isn't a nested indefinite? (Check that it wasn't `ReadStartByteString(); ReadByteString();` and the second one therefore saw indefinite inside indefinite)

Reply:
This is checked in the `PeekInitialByte()` method. There's also a test that verifies this.